samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -720,7 +722,7 @@ def resolve_sub_attribute(intrinsic_item, symbol_resolver):
             )
             result = resolve_sub_attribute(sanitized_item, self._symbol_resolver)
             sub_str = re.sub(
-                pattern=r"\$\{" + sub_item + r"\}", string=sub_str, repl=result
+                pattern=r"\$\{" + sub_item + r"\}", string=sub_str, repl=str(result)
```

Comment:
Because Python expects the replacement to be a string. In some cases customers provide an integer value via say a CFN parameter like az number or something.